File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/BugReport.java
package de.tum.in.tumcampusapp.models.tumcabe;

import android.content.Context;

import de.tum.in.tumcampusapp.auxiliary.NetUtils;
import de.tum.in.tumcampusapp.trace.G;
import de.tum.in.tumcampusapp.trace.Util;

public class BugReport {

    private final String packageName;
    private final String packageVersion;
    private final String packageVersionCode;
    private final String phoneModel;
    private final String androidVersion;
    private final String networkWifi;
    private final String networkMobile;
    private final String gps;
    private final String screenWidth;
    private final String screenHeight;
    private final String screenOrientation;
    private final String screenDpi;
    private final String stacktrace;
    private final String log;

    public BugReport(Context c, String stacktrace, String log) {
        super();
        this.packageName = G.appPackage;
        this.packageVersion = G.appVersion;
        this.packageVersionCode = String.valueOf(G.appVersionCode);

        this.phoneModel = G.phoneModel;
        this.androidVersion = G.androidVersion;

        this.networkWifi = NetUtils.isConnectedWifi(c) ? "true" : "false";
        this.networkMobile = NetUtils.isConnectedMobileData(c) ? "true" : "false";
        this.gps = Util.isGPSOn(c);

        String[] screenProperties = Util.getScreenProperties(c);
        this.screenWidth = screenProperties[0];
        this.screenHeight = screenProperties[1];
        this.screenOrientation = screenProperties[2];
        this.screenDpi = screenProperties[3] + ":" + screenProperties[4];

        this.stacktrace = stacktrace;
        this.log = log;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/BugReport.kt
package de.tum.`in`.tumcampusapp.models.tumcabe

import android.content.Context

import de.tum.`in`.tumcampusapp.auxiliary.NetUtils
import de.tum.`in`.tumcampusapp.trace.G
import de.tum.`in`.tumcampusapp.trace.Util

data class BugReport(var packageName: String = "",
                     var packageVersion: String = "",
                     var packageVersionCode: String = "",
                     var phoneModel: String = "",
                     var androidVersion: String = "",
                     var networkWifi: String = "",
                     var networkMobile: String = "",
                     var gps: String = "",
                     var screenWidth: String = "",
                     var screenHeight: String = "",
                     var screenOrientation: String = "",
                     var screenDpi: String = "",
                     var log: String = "",
                     var stacktrace: String = "") {

    companion object {
        fun getBugReport(c: Context, stacktrace: String, log: String): BugReport {
            val screenProperties = Util.getScreenProperties(c)
            return BugReport(
                    packageName = G.appPackage,
                    packageVersion = G.appVersion,
                    packageVersionCode = G.appVersionCode.toString(),
                    phoneModel = G.phoneModel,
                    androidVersion = G.androidVersion,
                    networkWifi = NetUtils.isConnectedWifi(c).toString(),
                    networkMobile = NetUtils.isConnectedMobileData(c).toString(),
                    gps = Util.isGPSOn(c),
                    screenWidth = screenProperties[0],
                    screenHeight = screenProperties[1],
                    screenOrientation = screenProperties[2],
                    screenDpi = "${screenProperties[3]}:${screenProperties[4]}",
                    stacktrace = stacktrace,
                    log = log
            )
        }
    }
}
